config/default.yml

```diff
@@ -49,6 +49,12 @@ Rails/ActionFilter:
   Include:
     - app/controllers/**/*.rb
 
+Rails/ActiveModelErrorsDirectManipulation:
+  Description: 'Avoid manipulating ActiveModel errors hash directly.'
+  Enabled: pending
+  Safe: false
```

Comment:
Just out of curiosity – is there a good reason to make this cop unsafe? There is a possibility that it detects non-AR statements, but I haven't seen such code in practice.

Going back to the question I forgot to answer:
> Would you happen to know how to rubocop into git submodules' content?

You `git checkout --include-submodules`, and then `fd --hidden --no-ignore .rubocop | xargs rm`. I don't know a better way to skip submodule's RuboCop configuration.

But usually GitLab's source is a good enough sandbox.

Reply:
@pirj Yes, non-AR classes with `errors` method is what I fear, would that qualify as "unsafe"? We don't have those in GitLab codebase, but I feel `errors` is generic enough that it can happen in utility classes.

---

Thanks. I found that there are some false positives on non-manipulative methods:

```log
redmine/app/controllers/wiki_controller.rb:74:10: C: Rails/ActiveModelErrorsDirectManipulation: Avoid manipulating ActiveModel errors as hash directly.
      if @page.errors[:title].blank?
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

This happens after the refactoring. I might be able to take a look tomorrow (and adding more tests). So please hold off from merging for now. Thanks!

Reply:
@pirj I've added specs and fixed the issue:

1. The `MANIPULATIVE_METHODS` was a single element Set (the element being an array). I splat it now.
2. The [constant seems to be references without `#`](https://docs.rubocop.org/rubocop-ast/node_pattern.html#const-or-const-for-constants) so I removed it.

Spec seems to pass so I guess all is good?